pump/storage/storage.go

```diff
@@ -764,18 +778,18 @@ func (a *Append) doGCTS(ts int64) {
 				deletedKv.Add(float64(batch.Len()))
 				batch.Reset()
 			}
-			log.Info("Finish KV GC", zap.Int64("ts", ts), zap.Int("delete num", deleteNum))
-			break
+			alreadyGcTS = endTS
```

Comment:
This line is a little tricky, although this is extracted as an independent function, this line assumes that the returned value is used in the `alreadyGcTS >= ts` check. If someone later uses the returned value and expects it to be the real last ts, he would be wrong.

Reply:
why not put L787 ～ L790 before L772？

Reply:
the mean of `alreadyGcTS` is that the KVs before this TS are already removed. and it's not mean that this TS is the last one that was removed. 
this line `alreadyGcTS = endTS` is correct, because this function can promise that all KVs smaller than `alreadyGcTS` will be deleted

Reply:
It's correct, but tricky. You might just explicitly add another return value to dictate to the caller whether there's more to be deleted.